src/user/mod.rs

```diff
@@ -1,28 +1,58 @@
+pub mod rust;
+
 use crate::parser::{self, NixNode};
 
 use std::{collections::HashMap, io::Cursor, str::FromStr};
 
 use parse_display::{Display, FromStr};
 use skim::prelude::*;
+use smol_str::SmolStr;
 
-//#[derive(Eq, PartialEq, Debug, Clone)]
-//pub struct UserResult {
-//user_selection: String,
-//}
+#[derive(Debug, Default, Clone, PartialEq, Eq, Display)]
+pub struct SmlStr(pub SmolStr);
 
-#[derive(Eq, PartialEq, Debug, Clone)]
-pub struct UserMetadata {
-    pub root: Option<NixNode>,
-    pub inputs: Option<HashMap<String, NixNode>>,
-    pub filename: Option<String>,
+impl SmlStr {
+    pub const fn new_inline(s: &str) -> Self {
+        Self(SmolStr::new_inline(s))
+    }
 }
 
-impl UserMetadata {
-    pub fn root_ref(&self) -> &NixNode {
-        self.root.as_ref().unwrap()
+impl Into<String> for SmlStr {
+    fn into(self) -> String {
+        self.0.into()
+    }
+}
+
+impl From<String> for SmlStr {
+    fn from(s: String) -> Self {
+        Self(s.into())
+    }
+}
+
+impl From<&String> for SmlStr {
+    fn from(s: &String) -> Self {
+        Self(s.into())
     }
+}
+
+impl FromStr for SmlStr {
```

Comment:
Is this the only reason?

Reply:
Yes. SmolStr was apparently used already, so I just made a wrapper around it.

Reply:
This should be clarified in a doc-comment above `struct SmlStr`.